### DISCUSSION TITLE
Dry-Events

### DISCUSSION THREAD

**Discussion by btihen** [hanamimastery/episodes#7] [Suggestions]
Can you give an example with dry-events (and its pub/sub) aspects and why add that beyond just using dependency injection is helpful (or at least when and why its helpful)

Thanks for awesome content.

PS - when you have enough videos completed (and Hanami 2 is released) - I hope you consider organizing the transcripts into a book

PPS - I think its beyond the scope of your site / mission, but the dry-rb gems are quite cool & added / extra content of using them in rails would also be cool

**Comment by swilgosz** [comment]
> PPS - I think its beyond the scope of your site / mission, but the dry-rb gems are quite cool & added / extra content of using them in rails would also be cool

@btihen I actually do release some rails-based episodes from time to time, but it's not a focus for me at the moment. It's more likely that I will do sth like this in occasion of guest-blogging on other people's blogs. The other scenario is, when other people would like to submit their content to Hanami Mastery and I can the help in making a screencast out of such article :). It's OSS, everyone can contribute:).

**Reply by btihen** [reply]
I have a colleague who has done a little dry-gems in rails - I'll look at what he's done and see if it can built upon.

**Comment by swilgosz** [comment]
> PS - when you have enough videos completed (and Hanami 2 is released) - I hope you consider organizing the transcripts into a book

It's awesome to see you would be interested in such thing. Yes, I do plan to write e-books first, online courses etc... Just need some time and probably partner with someone :).

**Reply by btihen** [reply]
I'd be glad to help / partner.  I am hoping to use your videos and build a 'fun' app that's just complex enough to explore the features of Hanami and its value add.  Is something like this helpful and interesting?

**Comment by swilgosz** [comment]
> Can you give an example with dry-events (and its pub/sub) aspects and why add that beyond just using dependency injection is helpful (or at least when and why its helpful)

I do plan to cover ALL DRY, ROM and Hanami libs at some point. `dry-effects` is sth I have never used myself yet, but I think it may be used in hanami for monitoring.

I will cover it at some point, just I'm uncertain about priorities. I would love if more people would show their interest and the I will push it forward for sure. :) . Thanks for the ideas! ❤️

**Reply by btihen** [reply]
We do a lot of background jobs - mostly with resque, but at the moment when its basically fire and forget.  Mostly good, but we use a lot of afterhooks so things are tightly coupled and when they fail, we don't always have a way to guarantee delivery and in some cases we need that - so we are migrating important use cases to Rabbit MQ.  

I haven't looked at what dry-events offers, but I though a low overhead event handler might be a nice middle ground - espescially when communicating between modules (which we are considering moving toward), but as I understand these are native to Hanami.
